File: src/main/java/com/projectswg/holocore/resources/support/global/commands/callbacks/chat/CmdBroadcast.java
package com.projectswg.holocore.resources.support.global.commands.callbacks.chat;

import com.projectswg.holocore.intents.support.global.chat.SystemMessageIntent;
import com.projectswg.holocore.resources.support.global.commands.ICmdCallback;
import com.projectswg.holocore.resources.support.global.player.Player;
import com.projectswg.holocore.resources.support.objects.swg.SWGObject;
import com.projectswg.holocore.resources.support.objects.swg.creature.CreatureObject;
import org.jetbrains.annotations.NotNull;

public final class CmdBroadcast implements ICmdCallback {
	
	@Override
	public void execute(@NotNull Player player, SWGObject target, @NotNull String args) {
		CreatureObject creature = player.getCreatureObject();
		if (creature == null)
			return;
		SystemMessageIntent.broadcastPlanet(creature.getTerrain(), args);
	}
	
}

File: src/main/java/com/projectswg/holocore/resources/support/global/commands/callbacks/chat/CmdBroadcast.kt
package com.projectswg.holocore.resources.support.global.commands.callbacks.chat

import com.projectswg.holocore.intents.support.global.chat.SystemMessageIntent.Companion.broadcastPlanet
import com.projectswg.holocore.resources.support.global.commands.ICmdCallback
import com.projectswg.holocore.resources.support.global.player.Player
import com.projectswg.holocore.resources.support.objects.swg.SWGObject

class CmdBroadcast : ICmdCallback {
	override fun execute(player: Player, target: SWGObject?, args: String) {
		val creature = player.creatureObject ?: return
		broadcastPlanet(creature.terrain, args)
	}
}
